runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -123,8 +133,12 @@ public void serialize(ByteBuf b) {
     @Override
     public void setGlobalAddress(long address) {
         super.setGlobalAddress(address);
-        this.getEntryMap().values().forEach(x -> {
-            x.setGlobalAddress(address);
-        });
+        this.entryMap.values()
```

Comment:
I think this is incorrect. 

This method is called after `getPayload`, but at that point not all SMR entries are deserialized, so by the time `getSMRUpdates` the entries won't have a global address.

Reply:
Maybe we can push the `setGlobalAddress` call down to `StreamViewAdapter`, instead of in `getPayload()`, to avoid this issue, since `StreamViewAdapter` will call `getSMRUpdates(id)` and finish deserialization.

Reply:
Also this issue might not be caught in unit tests, because unit test framework does not serialize the payload into data array and deserialize, it's just the payload being copied to server component, so the address i actually set in `useToken()` method before sending to sever, when getting back data from server and calling `getPayload()`, it won't invoke `((LogEntry) actualValue).setGlobalAddress(getGlobalAddress())`. We might need to write a specific unit test or use IT to reveal the problem.